plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java

```diff
@@ -0,0 +1,587 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.ml.rest;
+
+import static org.opensearch.ml.utils.TestHelper.makeRequest;
+import static org.opensearch.ml.utils.TestHelper.toHttpEntity;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.message.BasicHeader;
+import org.junit.Before;
+import org.opensearch.client.Response;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.ml.common.MLTaskState;
+import org.opensearch.ml.utils.TestHelper;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public class RestMLRAGSearchProcessorIT extends RestMLRemoteInferenceIT {
+
+    private static final String OPENAI_KEY = System.getenv("OPENAI_KEY");
+    private static final String OPENAI_CONNECTOR_BLUEPRINT = "{\n"
+        + "    \"name\": \"OpenAI Chat Connector\",\n"
+        + "    \"description\": \"The connector to public OpenAI model service for GPT 3.5\",\n"
+        + "    \"version\": 2,\n"
+        + "    \"protocol\": \"http\",\n"
+        + "    \"parameters\": {\n"
+        + "        \"endpoint\": \"api.openai.com\",\n"
+        + "        \"model\": \"gpt-3.5-turbo\",\n"
+        + "        \"temperature\": 0\n"
+        + "    },\n"
+        + "    \"credential\": {\n"
+        + "        \"openAI_key\": \""
+        + OPENAI_KEY
+        + "\"\n"
+        + "    },\n"
+        + "    \"actions\": [\n"
+        + "        {\n"
+        + "            \"action_type\": \"predict\",\n"
+        + "            \"method\": \"POST\",\n"
+        + "            \"url\": \"https://${parameters.endpoint}/v1/chat/completions\",\n"
+        + "            \"headers\": {\n"
+        + "                \"Authorization\": \"Bearer ${credential.openAI_key}\"\n"
+        + "            },\n"
+        + "            \"request_body\": \"{ \\\"model\\\": \\\"${parameters.model}\\\", \\\"messages\\\": ${parameters.messages}, \\\"temperature\\\": ${parameters.temperature} }\"\n"
+        + "        }\n"
+        + "    ]\n"
+        + "}";
+
+    private static final String AWS_ACCESS_KEY_ID = System.getenv("AWS_ACCESS_KEY_ID");
+    private static final String AWS_SECRET_ACCESS_KEY = System.getenv("AWS_SECRET_ACCESS_KEY");
+    private static final String AWS_SESSION_TOKEN = System.getenv("AWS_SESSION_TOKEN");
```

Comment:
@austintlee I think root problem isn't enabling access for Bedrock in the account. I highly doubt if we have these variables in our github environment.

@peterzhuamazon could you please help us to confirm this?

Reply:
That is strange...  If you look at my very first comment above, I grabbed that from the run that was kicked off upon my first push.  But I no longer see that under `Run aws-actions/configure-aws-credentials@v2`.  Maybe what I saw before was 'v1'?  But this step does indicate that aws credentials are being set up, no?

Reply:
@austintlee I figured out the issue. Yeah, we don't have these variables so it's not quite connecting with any aws resources. We need to setup a IAM role with the permission with bedrock and store that in the github secret. Security token is not scalable as this will expire eventually. I will work with @peterzhuamazon to set this up and update you on this. Thanks.

Reply:
I synced up with @gaiksaya from infra. The AWS creds which are available in CI are only for publishing tasks and its central to opensearch.org project[1] and they are not available for other tasks. 

Next Steps:
- We should setup another test account with scoped down role for Bedrock tests. 
- Also work with infra to figure out how to run these integration tests with Bedrock or 3P in release distribution tests.